### DISCUSSION TITLE
how to display a fallback logo when the one set in config isn't supported by terminal?

### DISCUSSION THREAD

**Discussion by GrassBlock1** [fastfetch-cli/fastfetch#1109] [Q&A] [answered]
I used fastfetch in my fish motd and the logo is a png image like this:
![图片](https://github.com/user-attachments/assets/8817f580-f686-47e2-85dc-fb5e9222796a)

It's supported by kitty, but when I switched to terminals like alacritty and the one in IntelliJ idea(which seems to be xterm-256color) and the image logo are not supported:
![图片](https://github.com/user-attachments/assets/9b7e8f43-c1b5-4a7c-852a-878dd56b035c)

And this is my config for logos in fastfetch:
```json
"logo": {
        "type": "auto",
        "source": "/home/grassblock/Pictures/1017124-new.png",
        "height": 6
    }
```
and my greeting section `config.fish`:
```fish
# greeting
function fish_greeting
  echo ""
  fastfetch -c $__fish_config_dir/motd.jsonc #this stands for the config above
  echo -e "\nWelcome to fish@$TERM, $USER!"
end
# greeting end
```

So is there any way to easily set a logo as a fallback? I had searched issues and discussions and seems there's no similar situations for this.

thanks in advance.

**Comment by CarterLi** [comment, marked as answer]
```fish
function fish_greeting
    if [ $TERM = xterm-kitty ]
        fastfetch -c $__fish_config_dir/motd.jsonc
    else
        fastfetch -c $__fish_config_dir/motd.jsonc --logo-type builtin
    end
end
```

**Reply by GrassBlock1** [reply]
thanks for quick reply! I tried this in my `config.fish` and it works.
But I still have two questions:
1. Is there any way to tell fastfetch to use a `small` logo via the command line options in this circumstance? I tried `--logo-type small` but it seems doesn't work.
2. Some terminals like konsole report themselves in $TERM as `xterm-256color`,but they do support display images: 
![图片](https://github.com/user-attachments/assets/5dcb0f12-e06b-4c3b-a434-552aed488191)
Is there any solutions for them?

**Reply by CarterLi** [reply]
> I tried `--logo-type small` but it seems doesn't work.

Try `-l arch-small`

**Reply by GrassBlock1** [reply]
> > I tried `--logo-type small` but it seems doesn't work.
> 
> Try `-l arch-small`

I don't think it works....(the config file was copied as-is, from the original one)
![图片](https://github.com/user-attachments/assets/23c411c5-2454-48b3-bde5-f2c35640b38d)
